addons/account_audit_trail/models/mail_message.py

```diff
@@ -175,7 +175,19 @@ def _search_audit_log_related_record_id(self, model, operator, value):
     def _except_audit_log(self):
         if self.env.context.get('bypass_audit') is bypass_token:
             return
-        for message in self:
+        to_check = self
+        partner_message = self.filtered(lambda m: m.model == 'res.partner')
+        if partner_message:
+            # The audit trail uses the cheaper check on `customer_rank`, but that field could be set
+            # without actually having an invoice linked (i.e. creation of the contact through the
+            # Invoicing/Customers menu)
+            moves_with_partners = self.env['account.move'].sudo().search([
+                ('partner_id', 'in', sorted(set(partner_message.mapped('res_id')))),
+                ('company_id.check_account_audit_trail', '=', True),
+            ])
+            if not moves_with_partners:
```

Comment:
We could raise in a else here directly then, no need to get the other computes.

Reply:
It's fine with my latest change I think, we won't recompute the value for these messages